packages/flutter/lib/src/material/predictive_back_page_transitions_builder.dart

```diff
@@ -37,6 +37,47 @@ class PredictiveBackPageTransitionsBuilder extends PageTransitionsBuilder {
   /// predictive back transition.
   const PredictiveBackPageTransitionsBuilder();
 
+  @override
+  Widget buildTransitions<T>(
+      PageRoute<T> route,
+      BuildContext context,
+      Animation<double> animation,
+      Animation<double> secondaryAnimation,
+      Widget child,
+      ) {
+    return _PredictiveBackGestureDetector(
+      route: route,
+      builder: (BuildContext context, _, __) {
+        // Only do a predictive back transition when the user is performing a
+        // pop gesture. Otherwise, for things like button presses or other
+        // programmatic navigation, fall back to ZoomPageTransitionsBuilder.
+        if (route.popGestureInProgress) {
+          return _PredictiveBackPageFullScreenTransition(
```

Comment:
I think PredictiveBackPageTransitionBuilder should probably do the shared element transition, since that seems to be the default on native Android? Then we would have a separate PredictiveBackFullScreenPageTransitionBuilder for that transition. Or we could even delete it... Do we know if the full screen transition exists on native Android? Like can I choose to use it when writing a native Android app? It's hard to tell since everything is in beta still.